src/main/java/com/arckenver/nations/listener/BuildPermListener.java

```diff
@@ -16,17 +16,15 @@
 import com.arckenver.nations.ConfigHandler;
 import com.arckenver.nations.DataHandler;
 import com.arckenver.nations.LanguageHandler;
+import org.spongepowered.api.world.World;
 
 public class BuildPermListener
 {
 
 	@Listener(order=Order.FIRST)
 	public void onPlayerPlacesBlock(ChangeBlockEvent.Place event, @First Player player)
 	{
-		if (!ConfigHandler.getNode("worlds").getNode(event.getTargetWorld().getName()).getNode("enabled").getBoolean())
-		{
-			return;
-		}
+
```

Comment:
Since according to https://github.com/SpongePowered/SpongeAPI/commit/403a0d8c1d5684c1e6b2f6505f7e3cb1883e9bb0 it is possible that one event contains multiple block changes at ones, which do not have to be in the same world. (No idea why)
Because you are looping through all transactions afterwards, I decided to move the 'enabled world' check to that loop to test it for each transaction (possibly in a different world).

I did apply this behavior to all changed events except for the entity spawn event as you only check the permission for the first entity anyway